openedx/core/djangoapps/credentials/management/commands/notify_credentials.py

```diff
@@ -198,6 +200,7 @@ def send_notifications(self, certs, grades, site_config=None, delay=0, page_size
 
         # First, do certs
         for i, cert in paged_query(certs, delay, page_size):
+            cert = GeneratedCertificate.objects.get(id=cert.id)
```

Comment:
So I don't quite understand. This is basically telling the Python interpreter that it can clean up the cert from the iteration? And that frees up the cached memory?

How is that different from the `cert` object being cleaned up by Python at the ned of the `for` loop iteration?

Speaking of iteration... Have we looked at whether the `iterator()` call would help here? That seems related to this issue. And I think is a feature I didn't know about when I wrote this management command.

 https://docs.djangoproject.com/en/1.11/ref/models/querysets/#iterator

Reply:
@mikix Yes object would't be cached because its not evaluated by the queryset. We are making a separate call to DB and object would be destroyed once it goes out of scope.
Django iterator does the same thing. So I have used the `iterator()`.